packages/truffle-db/src/artifacts/schema.ts

```diff
@@ -75,5 +96,61 @@ export const schema = mergeSchemas({
       },
       
     },
+
+    Contract: {
+      name: {
+        fragment: "... on ContractObject { name: contractName }"
+      }, 
+      sourceContract: {
+        fragment: 
+        `... on ContractObject { 
+          ast
+        }`,
+        resolve: (obj, ast) => {
+          const { name, source } = obj;
+          const sourceContract = {
+            name: name, 
+            source: source, 
+            ast: ast
+          }
+          return sourceContract;
+        }
+      }, 
+    },
+
+    ContractConstructor: {
+      contract: {
+        fragment: 
+        `... on ContractObject { 
+          contents,
+          sourcePath,
+          ast
+        }`,
+        resolve: (obj, contents, sourcePath, ast) => {
+          const { name } = obj;
+          const contract = {
+            name: name,
+            sourceContract: {
+              name: name,
```

Comment:
Definitely prefer this new syntax wherever we can:

```suggestion
              name,
```